solr/solrj/src/java/org/apache/solr/common/params/ModifiableSolrParams.java

```diff
@@ -175,6 +177,18 @@ public boolean remove(String name, String value) {
   // ----------------------------------------------------------------
   // ----------------------------------------------------------------
 
+  public static void setShardAttributesToParams(
+      int purpose, String shard, ModifiableSolrParams params) {
```

Comment:
subjective: method could be non-static maybe and/or to add test coverage e.g. in `ModifiableSolrParamsTest.java`
```suggestion
  public void setShardAttributes(
      int purpose, String shard) {
```

Reply:
true, I did not realize an instance of ModifiableSolrParams is present at the place where the method is called.
I have added a unit test validating the code, please have a look.
Also I have made two changes to the method, please have a look at me know if this makes sense to you.
Thanks!